### PR TITLE
Check if the path exists.

### PR DESCRIPTION
**Pull request by ZePan110**
## Description

The summary of the proposed changes as long as the relevant motivation and context.

## Issues

List the issue or RFC link this PR is working on. If there is no such link, please mark it as `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

**Review by daisy-ycguo — COMMENTED** [review]
I wonder why this workflow belongs to GenAIExamples not GenAICompos.

**Comment by chensuyue** [comment]
> I wonder why this workflow belongs to GenAIExamples not GenAIComps.

We propose to setup checking workflow on Both GenAIExamples and GenAIComps. In GenAIExamples, we mainly checking for typos, in GenAIComps we mainly checking for Dockerfile path change in PR.

**Comment by daisy-ycguo** [comment]
> > I wonder why this workflow belongs to GenAIExamples not GenAIComps.
> 
> We propose to setup checking workflow on Both GenAIExamples and GenAIComps. In GenAIExamples, we mainly checking for typos, in GenAIComps we mainly checking for Dockerfile path change in PR.

I think this workflow is to checking for Dockerfile path in GenAICompos. Correct me if I'm wrong.

**Comment by chensuyue** [comment]
> > > I wonder why this workflow belongs to GenAIExamples not GenAIComps.
> > 
> > 
> > We propose to setup checking workflow on Both GenAIExamples and GenAIComps. In GenAIExamples, we mainly checking for typos, in GenAIComps we mainly checking for Dockerfile path change in PR.
> 
> I think this workflow is to checking for Dockerfile path in GenAICompos. Correct me if I'm wrong.

This workflow is checking the Dockerfile path in GenAIExamples readme can be found correctly in GenAIComps, to avoid typos. 
We will add another workflow in GenAIComps PR to make sure the Dockerfile path change can be update parallel in GenAIExamples.

File: .github/workflows/path_detection.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    - name: Clone repo GenAIComps
+      run: |
+        git clone https://github.com/opea-project/GenAIComps.git
+
+    - name: Check for missing Dockerfile paths in GenAIComps
+      run: |
+        grep -Ern 'docker build .* -f comps/.+/Dockerfile' --include='*.md' . | while IFS=: read -r file line content; do
+            dockerfile_path=$(echo "$content" | awk -F '-f ' '{print $2}' | awk '{print $1}')
+            if [[ ! -f "../GenAIComps/${dockerfile_path}" ]]; then
```
Comment (daisy-ycguo) [inline .github/workflows/path_detection.yml:27]: I wonder the folder `GenAICompos` cloned in previous step is a child of the folder of `GenAIExamples` or a sibling of folder `GenAiExamples`. If `GenAICompos` is the child of `GenAIExamples`, `../GenAIComps` is wrong. Please confirm. Thank you.

File: .github/workflows/path_detection.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          git clone https://github.com/opea-project/GenAIComps.git
+
+      - name: Check for missing Dockerfile paths in GenAIComps
+        run: |
+          cd ${{github.workspace}}
+          grep -Ern 'docker build .* -f comps/.+/Dockerfile' --include='*.md' . | while IFS=: read -r file line content; do
+              dockerfile_path=$(echo "$content" | awk -F '-f ' '{print $2}' | awk '{print $1}')
+              if [[ ! -f "../GenAIComps/${dockerfile_path}" ]]; then
+                  echo "Missing Dockerfile: ${dockerfile_path} (Referenced in ${file}:${line})"
+                  exit 1
```
Comment (chensuyue) [inline .github/workflows/path_detection.yml:34]: Let's finish check all the files, and raise issue at the end.

Comment (ZePan110) [reply .github/workflows/path_detection.yml]: Done.

**Review by chensuyue — APPROVED** [review]

**Review by daisy-ycguo — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**